src/Servant/QuickCheck/Internal/Predicates.hs

```diff
@@ -429,7 +430,8 @@ finishPredicates p req mgr = go `catch` \(e :: PredicateFailure) -> return $ Jus
   where
     go = do
      resps <- getRequestPredicate (requestPredicates p) req mgr
-     mapM_  (getResponsePredicate $ responsePredicates p) resps
+     let responder = getResponsePredicate (responsePredicates p) req
```

Comment:
Hmm, this isn't quite right, I think. The actual requests that resulted in the responses `resps` are not necessarily `req`. Rather, they are the requests generated by the `RequestPredicate`s *based on* `req`.

Reply:
I have to think a little more about this. It feels like it's becoming evident that the `Request`/`ResponsePredicate` distinction is somewhat artificial or misleading. Really what's at stake is something like the distinction between request generator (or generator modifiers) and (request + response) predicates. But separating those two too much is also wrong, since usually one wants to generate certain requests *in order to* check the responses.

Reply:
I am glad to have your perspective on this. I was wondering about the difference between `ResponsePredicate` and `RequestPredicate` when I was working on this PR, but I figured the distinction is still valid as a `RequestPredicate` needs a  `Manager` in order to actually issue requests.

I have to confess I didn't look closely enough at `finishPredicates`, so it's a relief that you caught my mistake, which I can see clearly now.